src/components/ImageWithSizeCalculation.tsx

```diff
@@ -1,66 +1,54 @@
-import _ from 'underscore';
 import React, {useState, useRef, useEffect} from 'react';
-import {View} from 'react-native';
-import PropTypes from 'prop-types';
+import {View, ViewStyle} from 'react-native';
+import delay from 'lodash/delay';
 import Log from '../libs/Log';
 import styles from '../styles/styles';
 import FullscreenLoadingIndicator from './FullscreenLoadingIndicator';
-import Image from './Image';
+import Image, {RESIZE_MODES} from './Image';
 
-const propTypes = {
+type ImageWithSizeCalculationProps = {
     /** Url for image to display */
-    url: PropTypes.string.isRequired,
+    url: string;
 
     /** Any additional styles to apply */
-    // eslint-disable-next-line react/forbid-prop-types
-    style: PropTypes.any,
+    style: ViewStyle;
 
     /** Callback fired when the image has been measured. */
-    onMeasure: PropTypes.func,
+    onMeasure: (args: {width: number; height: number}) => void;
 
     /** Whether the image requires an authToken */
-    isAuthTokenRequired: PropTypes.bool,
-};
-
-const defaultProps = {
-    style: {},
-    onMeasure: () => {},
-    isAuthTokenRequired: false,
+    isAuthTokenRequired: boolean;
 };
 
 /**
  * Preloads an image by getting the size and passing dimensions via callback.
  * Image size must be provided by parent via width and height props. Useful for
  * performing some calculation on a network image after fetching dimensions so
  * it can be appropriately resized.
- *
- * @param {Object} props
- * @returns {React.Component}
- *
  */
-function ImageWithSizeCalculation(props) {
-    const isLoadedRef = useRef(null);
+function ImageWithSizeCalculation({url, style, onMeasure, isAuthTokenRequired}: ImageWithSizeCalculationProps) {
+    const isLoadedRef = useRef<boolean | null>(null);
     const [isImageCached, setIsImageCached] = useState(true);
     const [isLoading, setIsLoading] = useState(false);
 
     const onError = () => {
-        Log.hmmm('Unable to fetch image to calculate size', {url: props.url});
+        Log.hmmm('Unable to fetch image to calculate size', {url});
     };
 
-    const imageLoadedSuccessfully = (event) => {
+    const imageLoadedSuccessfully = (event: {nativeEvent: {width: number; height: number}}) => {
```

Comment:
```suggestion
    const imageLoadedSuccessfully = (event: OnLoadEvent) => {
```

We can use `OnLoadEvent` from `react-native-fast-image`

Reply:
updated